gordo_components/builder/azure_utils.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Iterator, Tuple, Union
+from datetime import datetime
+from pytz import UTC
+import logging
+
+from mlflow.entities import Metric, Param
+from mlflow.tracking import MlflowClient
+from azureml.core import Workspace
+from azureml.core.authentication import (
+    InteractiveLoginAuthentication,
+    ServicePrincipalAuthentication,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_mlflow_client(azureml_config: dict = {}) -> MlflowClient:
+    """
+    Set remote tracking URI for mlflow to AzureML workspace
+
+    Parameters
+    ----------
+    azure_config: dict
+        AzureML Workspace configuration to use for remote MLFlow tracking. Passing only
+        Workspace parameters with a missing our empty `auth` key will result in
+        interactive authentication to AzureML.
+
+        Example::
+
+            `{
+                 "subscription_id":<value>,
+                 "resource_group":<value>,
+                 "workspace_name":<value>
+                 "auth": {
+                     "tenant_id":<value>,
+                     "service_principal_id":<value>,
+                     "service_principal_password":<value>
```

Comment:
I feel we must (unfortunately?) look at the Datalake authentication we already have. It appears that it is using service principal, with much the same data (tenant/id/secret). In a way it is good, but this "triplet" already comes in as an environment variable, so you could probably just re-use it.

Reply:
We already talked about this, but just in case any other perusers have additional thoughs: I'm using the `azureml.core.authentication.ServicePrincipalAuthentication` class for auth. Not sure if those are the same, but I'll that those secrets will work and report back here.